### DISCUSSION TITLE
Potential Issues Sending XDrip Treatments to Tidepool

### DISCUSSION THREAD

**Discussion by ballenstein** [NightscoutFoundation/xDrip#2900] [Q&A] [answered]
First, THANK YOU!!!!  Once we found and got xDrip+ setup to use with the Libre 3 in [Web Follower mode](https://xdrip.readthedocs.io/en/latest/install/webfollower/), it has definitely helped with the day-to-day management of our daughter's T1D on MDI.  The Libre 3 software has things to be desired for sure, so its nice to have a good companion tool like this.

We've experimented with the use of manual treatment entries in XDrip+ for tracking things like finger stick bg, carbs, bolus, & basal values (setup [multiple insulin types](https://xdrip.readthedocs.io/en/latest/use/predictions/#multiple-insulin-types)).  Entry on the master phone and XDrip sync to [following](https://xdrip.readthedocs.io/en/latest/install/xdripfollower/) phones works great for CGM **AND** treatment data...we can see all of this stuff on any one of the master or following phones.  

We've also setup the [Tidepool Cloud Upload](https://xdrip.readthedocs.io/en/latest/use/cloud/#tidepool) feature in hopes of making it simpler to track, analyze, and share most of our data, esp. with our medical provider (they already use Tidepool for some things).  All of the CGM data shows up in Tidepool, but my issue is that most of the treatments being entered in XDrip+ don't make it to Tidepool (*some* do though).  I've tried always using the master device for entry of treatments (where follow/sync works fine both ways), but I still don't see all treatments make it to Tidepool.  My expectation is that I would see any manually entered bg, carb, or insulin treatments transferred to Tidepool, but I only get them sporadically while seeing all CGM data make it up.  I did try to search for existing issues or discussions, but couldn't find anything that seemed like a match to the issue I'm describing here.  I also did ensure that the `Don't upload treatments` checkbox in the TIdepool setup was **NOT** checked....didn't seem like that could be it if some make it but others don't.

While not the primary issue here, its also worth mentioning, I typically see any basal entries that do make it to Tidepool showing up in the bolus category instead.  It does seem like these basal entries are uploaded a bit more consistently, kinda like the CGM data (this is likely already covered in [Issue 1158](https://github.com/NightscoutFoundation/xDrip/issues/1158) from Jan. 2020.).  

Does anyone have any ideas or know if I should open an issue for this?  Again, **THANK YOU** for the creation and use of this tool.  We greatly appreciate it.

**Comment by psonnera** [comment]
This is very strange indeed. Do you think it can be linked to the Data age setting (i.e. 15 minutes) and you were checking too soo, or looking in the past you still can't see them?

**Reply by ballenstein** [reply]
I do believe that this setting is set to the default of 15 minutes.  I've typically been checking for this after a day or two of entry, so always more than that 15 minutes threshold.  Tidepool doesn't have a mobile interface, so I'm logging into a computer later on vs all of the XDrip+ entry being done on the phone throughout the day (which is the experience I'm looking for).

**Comment by ballenstein** [comment]
Here is an example of the issue I'm describing using data from yesterday.

The xDrip `Glucose History` screenshot here is from a "following" handset, and includes treatment values for carb, insulin (both bolus and basal), and finger stick values:
<img src="https://github.com/NightscoutFoundation/xDrip/assets/16958695/1b84280e-29c2-41cf-b953-cf2d12d71cf2" width="500" />

As you can see here in Tidepool, the whole day's CGM values are here.  However, only the finger stick and insulin treatments from 8:25am and the basal insulin treatment (which appears to be categorized as a bolus instead as described in [Issue 1158](https://github.com/NightscoutFoundation/xDrip/issues/1158) 😞 ) at 8pm made it up.  None of the other treatments (3 other finger sticks, 3 other bolus insulin, and 3 carb treatments) show up in Tidepool.

<img src="https://github.com/NightscoutFoundation/xDrip/assets/16958695/ff7b1169-2569-4d7e-ac2a-773ac03deb2b" />

**Reply by psonnera** [reply]
@ballenstein why does the + sign between units and carbs show as a symbol in your xDrip+? I might be wrong but treatments without this symbol do upload correctly.

**Reply by ballenstein** [reply]
I don't know why the `+` is not showing up properly.  I could check that on our other phone I suppose. Those two treatments with the symbol you noticed in addition to the 2g & .5u treatments from that day didn't make it up to Tidepool, so it doesn't appear to me to be related to that...

**Comment by ballenstein** [comment]
Not sure why it took me this long, but I finally hooked up our BG meter (AccuChek Guide) to xDrip+ today over [Bluetooth](https://xdrip.readthedocs.io/en/latest/use/glucometer/#if-you-use-a-compatible-bluetooth-meter).  I can see lots of readings pulled into xDrip+ in history now (on the master and following devices) 👍 .  I only see the one finger poke from today so far in Tidepool and am hoping the rest from today forward will continue to auto-magically (with the help of technology) show up in Tidepool going forward 🤞 .

Given the change above, I was also thinking about giving the entry of carbs and insulin (on MDI) another go / test again to see if anything has changed, but also wanted to bump this discussion about the issues I was having previously with this.  Is uploading any treatments entered in XDrip+ to Tidepool a fully supported / implemented feature?  Should all entered treatments be uploaded?  Should I use the multiple insulin types, including basal entry, or just put in a note for basal for now?  If I find otherwise on this [Tidepool Upload](https://xdrip.readthedocs.io/en/latest/use/cloud/#tidepool) feature, should I [enter an Issue](https://github.com/NightscoutFoundation/xDrip/issues) instead of [this discussion](https://github.com/NightscoutFoundation/xDrip/discussions/2900)?

Just trying to round out our toolbox well for data entry / management and reduce the cognitive load and work around T1D in our life while doing MDI + CGM + BG Meter for our management currently.  **THANK YOU*** again for any assistance with this!!!

xDrip+ version 23ccbd1-2023.07.16

**Comment by ballenstein** [comment, marked as answer]
I have a feeling that this is due to a combination of factors, including the `Data Age` setting @psonnera suggested earlier.  If I bump this `Data Age` setting out longer, say 12 hours or 24 hours, then xDrip doesn't attempt to upload **ANY** data it has until 12 hours afterwards.  By this time (12 hours later) we have been able to enter any bolus or carb treatments for the day, so when xDrip tries to upload the CGM data that is **ALWAYS** there for timeframe `A to B` to Tidepool, it also has the other treatment data for that same timeframe to send along with it.  It seems like its walking a timestamp forward on a timeline to track these uploads, so if it has sent CGM data for that timeframe already, it will "miss" anything else for that same timeframe that might not have been entered in "realtime" like the CGM data is.

Not sure if there is anything that could be done to make this better.  I'm going to continue to try my work-around by setting the `Data Age` to something long enough that gives us time to enter any data from that day before the upload attempt to Tidepool happens.  We would love to see a way that we could enter treatments for any timeframe and see them sent to Tidepool just like they are sent to other following xDrip+ devices (especially if they could be entered for previous days if we need to catch up).
